mobile_verifier/src/speedtests.rs

```diff
@@ -99,577 +85,107 @@ impl SpeedtestDaemon {
         file: FileInfoStream<CellSpeedtestIngestReport>,
     ) -> anyhow::Result<()> {
         tracing::info!("Processing speedtest file {}", file.file_info.key);
-
         let mut transaction = self.pool.begin().await?;
-        let reports = file.into_stream(&mut transaction).await?;
-
-        let mut validated_speedtests = pin!(
-            SpeedtestRollingAverage::validate_speedtests(
-                &self.gateway_client,
-                reports.map(|s| s.report),
-                &mut transaction,
-            )
-            .await?
-        );
-        while let Some(speedtest) = validated_speedtests.next().await.transpose()? {
-            speedtest.write(&self.file_sink).await?;
-            speedtest.save(&mut transaction).await?;
+        let mut speedtests = file.into_stream(&mut transaction).await?;
+        while let Some(speedtest_report) = speedtests.next().await {
+            let pubkey = speedtest_report.report.pubkey.clone();
```

Comment:
Do you need this clone here? Looks like you only pass it by reference for the rest of the operation

Reply:
you can just pass by reference as `&speedtest_report.report.pubkey;`